### DISCUSSION TITLE
How to detach a poco thread?

### DISCUSSION THREAD

**Discussion by jhodgson2** [pocoproject/poco#3996] [General]
Hi, I've started using the poco library for a project and I'm wondering how I can detach a poco thread. I'm unable to find anything in the docs for this functionality. I need to be able to create threads which won't be joined and instead will send a notification back to the main thread once they're done.

Thank you!

**Comment by kblaschke** [comment]
You may have figured it out yourself already, but for anyone else asking this question:

If you use `Poco::Thread`, the underlying OS thread is always detached (using `pthread_detach()` on UNIX and `CloseHandle()` on Windows) when the Thread object is destroyed and the thread is still running and not joined. This is quite different from `std::thread` behavior, which will call `std::terminate` if it's still running, so you have to call `std::thread::detach()` (or join the thread) before destroying the object.

So you can just create a Poco::Thread object, start() it and then immediately destroy the object again. The thread will then continue to run until finished.
